client/libra-dev/include/data.h

```diff
@@ -0,0 +1,181 @@
+    bool delegated_key_rotation_capability;
+    bool delegated_withdrawal_capability;
+    struct LibraEventHandle sent_events;
+    struct LibraEventHandle received_events;
+};
+
+struct LibraP2PTransferTransactionArgument {
+    uint64_t value;
+    uint8_t address[32];
+};
+
+enum TransactionType {
+    PeerToPeer = 0,
+    Mint = 1,
+    Unknown = -1,
+};
+
+struct LibraTransactionPayload {
+    enum TransactionType txn_type;
+    struct LibraP2PTransferTransactionArgument args; //
+};
+
+struct LibraRawTransaction {
+    uint8_t sender[32];
+    uint64_t sequence_number;
+    struct LibraTransactionPayload payload;
+    uint64_t max_gas_amount;
+    uint64_t gas_unit_price;
+    uint64_t expiration_time_secs;
+};
+
+struct LibraSignedTransaction {
+    struct LibraRawTransaction raw_txn;
+    uint8_t public_key[32];
+    uint8_t signature[64];
+};
+
+struct LibraAccountKey {
+    uint8_t address[32];
+    uint8_t private_key[32];
+    uint8_t public_key[32];
+};
+
+enum LibraEventType {
+    SentPaymentEvent = 1,
+    ReceivedPaymentEvent = 2,
+    UndefinedEvent = -1,
+};
+
+struct LibraPaymentEvent {
+    uint8_t sender_address[32];
+    uint8_t receiver_address[32];
+    uint64_t amount;
+    uint8_t* metadata;
+    size_t metadata_len;
+};
+
+struct LibraEvent {
+    enum LibraEventType event_type;
+    // TODO: address
+    uint8_t module[255];
+    uint8_t name[255];
+    // TODO: type_params
+    struct LibraPaymentEvent payment_event_data;
+    // TODO: other type of event_data
+};
+
+/*!
+ * Decode LibraAccountResource from bytes in AccountStateBlob.
+ *
+ * @param[in] buf contains encoded bytes of AccountStateBlob
+ * @param[in] len is the length of the signed AccountStateBlob memory buffer.
+ * @param[out] caller allocated LibraAccountResource to write into.
+ *
+ * @returns status code, one of LibraAPIStatus
+*/
+enum LibraStatus libra_LibraAccountResource_from(const uint8_t *buf, size_t len, struct LibraAccountResource *out);
+
+/*!
+ *  Get serialized signed transaction from a list of transaction parameters
+ *
+ * To get the serialized transaction in a memory safe manner, the client needs to pass in a pointer to a pointer to the allocated memory in rust
+ * and call free on the memory address with `libra_free_bytes_buffer`.
+ * @param[in] expiration_time_secs is the time this TX remain valid, the format is unix timestamp.
+ * @param[out] buf is the pointer that will be filled with the memory address of the transaction allocated in rust. User takes ownership of pointer returned by *buf, which needs to be freed using libra_signed_transcation_free
+ * @param[out] len is the length of the signed transaction memory buffer.
+*/
+enum LibraStatus libra_SignedTransactionBytes_from(const uint8_t sender_private_key[32], const uint8_t receiver[32], uint64_t sequence, uint64_t num_coins, uint64_t max_gas_amount, uint64_t gas_unit_price, uint64_t expiration_time_secs, uint8_t** ptr_buf, size_t* ptr_len);
```

Comment:
nit: Here you use `type** name` instead of `type **name` which is used elsewhere.